internal/io.h

```diff
@@ -8,8 +8,128 @@
  *             file COPYING are met.  Consult the file for details.
  * @brief      Internal header for IO.
  */
+
+#define RB_IO_T
+
 #include "ruby/ruby.h"          /* for VALUE */
-#include "ruby/io.h"            /* for rb_io_t */
+#include "ruby/io.h"
+
+/**
+ * IO  buffers.   This  is  an implementation  detail  of  ::rb_io_t::wbuf  and
+ * ::rb_io_t::rbuf.  People don't manipulate it directly.
+ */
+PACKED_STRUCT_UNALIGNED(struct rb_io_internal_buffer {
+    /** Pointer to the underlying memory region, of at least `capa` bytes. */
+    char *ptr;                  /* off + len <= capa */
+
+    /** Offset inside of `ptr`. */
+    int off;
+
+    /** Length of the buffer. */
+    int len;
+
+    /** Designed capacity of the buffer. */
+    int capa;
```

Comment:
Curious as why unsigned int here

Reply:
I agree, it's odd. But not part of my design, it's what's copied.